agent/handlers/task_server_setup_test.go

```diff
@@ -43,6 +43,7 @@ import (
 	"github.com/aws/amazon-ecs-agent/agent/stats"
 	mock_stats "github.com/aws/amazon-ecs-agent/agent/stats/mock"
 	apieni "github.com/aws/amazon-ecs-agent/ecs-agent/api/eni"
+	mock_api2 "github.com/aws/amazon-ecs-agent/ecs-agent/api/mocks"
```

Comment:
nit: Are there more descriptive names we can be using here as opposed to `mock_api` and `mock_api2`?

Reply:
Actually I'm seeing that `mock_api` import is currently only needed for calling `NewMockECSClient` function. Didn't we move ECS Client to ecs-agent module as part of this PR: https://github.com/aws/amazon-ecs-agent/pull/3745 ? Is there any reason the `NewMockECSClient` function needs to remain in agent module?

Reply:
Good question! 

Yes, we moved ECS Client to ecs-agent module. However, IMO the interfaces in this file provide an abstraction for Agent subcomponents to access ECS APIs. The ECS Client in ecs-agent module implicitly implements these interfaces but does not depend on them. It's the Agent subcomponents that do. So, I think it's okay for the interfaces to live here as long as there is no subcomponent in ecs-agent that depends on them. The reason I am moving `ECSTaskProtectionSDK` interface to ecs-agent is because the task protection handlers that I will be adding to ecs-agent module will depend on it.

Reply:
I will think of a better name.

Reply:
Renamed to `mock_taskprotection`.

Reply:
Got it, thanks for explaining!